tests/testthat/testCheckGeneSymbols.R

```diff
@@ -0,0 +1,67 @@
+# testDummy.R
+#
+# Dummy tests as a template for unit testing in test-driven development
+# for the rete project. If you test code that writes information to log-files,
+# turn logging off for all tests except for one specific block in which
+# you verify and validate your logs. Do not use your own log-file but
+# rely on the global option taht is set in the setup block. We have good
+# code samples for most test you will need - if you are not sure where to find
+# them, ask. Do not reinvent the wheel, but write your code according to the
+# principles you find in the existing files.
+#
+# Structure your tests according to the three principles expressed below:
+#    test_that("parameter errors are correctly handled", { ... })
+#    test_that("a sane input gives an expected output", { ... })
+#    test_that("a corrupt input does not lead to corrupted output", { ...})
+
+
+context("checkGeneSymbols functions")
+
+
+# ==== BEGIN SETUP AND PREPARE =================================================
+OLOG <- as.character(getOption("rete.logfile"))   # save original logfile name
+logFileName(fPath = tempdir(), setOption = TRUE)  # make tempdir() the log dir
+NL <- .PlatformLineBreak()
+# ==== END SETUP AND PREPARE ===================================================
+
+test_that("isGeneSymbol identifies invalid input", {
+    # Input contains invalid input.
+    input <- c(TRUE, 3.14, data.frame(key=c(1,2)), c(1,2,3))
+    expect_error(isGeneSymbol())
+    expect_error(isGeneSymbol(c()))
+    expect_error(isGeneSymbol(NULL))
+    expect_error(isGeneSymbol(input))
+    expect_error(isGeneSymbol(TRUE))
+    expect_error(isGeneSymbol(3.14))
+    expect_error(isGeneSymbol(data.frame(key=c(1,2))))
+    expect_error(isGeneSymbol(c(1,2,3)))
+    expect_error(isGeneSymbol(NA))
```

Comment:
If you mean isGeneSymbol(NA), the problem with that is that:
```
> typeof(NA)
[1] "logical"
> class(NA)
[1] "logical"
> mode(NA)
[1] "logical"
```
So should I explicitly code so that if the input it NA, then return `FALSE`?